src/KOKKOS/atom_vec_kokkos.cpp

```
@@ -845,14 +1375,1530 @@ void AtomVecKokkos::unpack_reverse_kokkos(const int &n,
+          _nspecial(i,1) = _nspecial(j,1);
+          _nspecial(i,2) = _nspecial(j,2);
+          for (int k = 0; k < _nspecial(j,2); k++)
+            _special(i,k) = _special(j,k);
+        }
+
+        if (_datamask & MU_MASK) {
+          _mu(i,0) = _mu(j,0);
+          _mu(i,1) = _mu(j,1);
+          _mu(i,2) = _mu(j,2);
+          _mu(i,3) = _mu(j,3);
+        }
+
+        if (_datamask & SP_MASK) {
+          _sp(i,0) = _sp(j,0);
+          _sp(i,1) = _sp(j,1);
+          _sp(i,2) = _sp(j,2);
+          _sp(i,3) = _sp(j,3);
+        }
+
+        if (_datamask & RADIUS_MASK)
+          _radius(i) = _radius(j);
+
+        if (_datamask & RMASS_MASK)
+          _rmass(i) = _rmass(j);
+
+        if (_datamask & OMEGA_MASK) {
+          _omega(i,0) = _omega(j,0);
+          _omega(i,1) = _omega(j,1);
+          _omega(i,2) = _omega(j,2);
+        }
+
+        // DPD-REACT package
+
+        if (_datamask & DPDTHETA_MASK) {
+           _dpdTheta(i) = _dpdTheta(j);
+          _uCond(i) = _uCond(j);
+          _uMech(i) = _uMech(j);
+          _uChem(i) = _uChem(j);
+          _uCG(i) = _uCG(j);
+          _uCGnew(i) = _uCGnew(j);
+        }
+      }
+    }
+  }
+};
+
+/* ---------------------------------------------------------------------- */
+
+int AtomVecKokkos::pack_exchange_kokkos(const int &nsend,DAT::tdual_double_2d_lr &k_buf,
+                                                 DAT::tdual_int_1d k_sendlist,
+                                                 DAT::tdual_int_1d k_copylist,
+                                                 ExecutionSpace space)
+{
+  atomKK->sync(space,datamask_exchange);
+  set_size_exchange();
+
+  if (nsend > (int) (k_buf.view_host().extent(0)*
+              k_buf.view_host().extent(1))/size_exchange) {
+    int newsize = nsend*size_exchange/k_buf.view_host().extent(1)+1;
+    k_buf.resize(newsize,k_buf.view_host().extent(1));
+  }
+
+  if (space == HostKK) {
+    if (size_exchange == 11) {
```

Maybe I'm missing something obvious, but why 11?

---

@jtclemm Good question. It is the size of the "default" exchange, which is 3 velocity + 3 positions + 1 id + 1 mask + 1 type + 1 image + 1 extra for buffer = 11, see https://github.com/lammps/lammps/blob/develop/src/atom_vec.cpp#L41-L42. However this could change so I removed the hardcoded value in the last commit.

---

Ah, I didn't get the +1 buffer. Makes sense not to hardcode though.